syntax = "proto3";

package marioai;

option java_package = "de.lmu.parl.proto";
option java_outer_classname = "MarioProtos";

message MarioMessage {
  enum Type { 
    INIT = 0;
    ACTION = 1;
    STATE = 2;
  }
  Type type = 1;
  Init init = 2;
  Action action = 3;
  State state = 4;
}

message Init {
  int32 difficulty = 1;
  int32 seed = 2;
  int32 r_field_w = 3;
  int32 r_field_h = 4;
  int32 level_length = 5;
}

message Action {
  int32 action_number = 1;
}

message State {
  int32 state = 1; // TODO
}